integrations/jdbc/jdbc/src/main/java/io/helidon/integrations/jdbc/ConditionallyCloseableConnection.java

```diff
@@ -43,14 +67,55 @@ public class ConditionallyCloseableConnection extends DelegatingConnection {
 
 
     /**
-     * Whether or not the {@link #close()} method will actually close
-     * this {@link DelegatingConnection}.
+     * A {@link SQLRunnable} representing the logic run by the {@link #checkOpen()} method.
+     *
+     * <p>This field is never {@code null}.</p>
+     *
+     * <p>This field is set based on the value of the {@code strictClosedChecking} argument supplied to the {@link
+     * #ConditionallyCloseableConnection(Connection, boolean, boolean)} constructor. It may end up deliberately doing
+     * nothing.</p>
+     *
+     * @see #ConditionallyCloseableConnection(Connection, boolean, boolean)
+     */
+    private final SQLRunnable closedChecker;
```

Comment:
Ha; I went through some of these mental gymnastics while writing this. I think I'm going to leave it. But for posterity here was the reasoning:

Each "business method" has a `checkOpen()` call.  The `checkOpen()` ideally takes no action, because hopefully you aren't whaling away on your `Connection` while it's closed. So `checkOpen()` checks to make sure the connection is open.  I named it this way because when you're scanning through this class, you see: "aha, yes, check to make sure the connection is open, and then forward the `setShardingKey` (for example) operation on to the delegate".

Next, thanks to backwards compatibility concerns, `checkOpen` should do _exactly_ nothing when "old" constructors are used.  Bad, but existing behavior, so….

But when "new" constructors are used, `checkOpen` should indeed cause failure when things are closed.

I didn't want all this if/then checking happening on each call, so I elected to swap out the "body" of the `checkOpen` method at construction time depending on whether you go with the old constructors or the new ones.

Now you know!  Thanks for reviewing.